custom_components/google_home/api.py

```diff
@@ -385,7 +315,8 @@ async def post(
 
         url = self.create_url(device.ip_address, PORT, endpoint)
 
-        HEADERS[HEADER_CAST_LOCAL_AUTH] = device.auth_token
+        headers = cast(Dict[str, str], HEADERS.copy())
```

Comment:
`HEADERS` is a shared const which shouldn't be modified. Modifying it actually introduces a race condition, it other thread will also modify it, you may end up with incorrect value. That's why you should create a copy.

To prevent const modifying I added restricted type and copy gets it as well. To bypass that I need to cast copied value to `Dict[str, str]` which can be modified.

Reply:
Ok, let's just remove `HEADERS` const. It only used here, so we can construct it when it's needed.